sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2471,6 +2471,24 @@ object functions {
   def second(e: Column): Column = withExpr { Second(e.expr) }
 
   /**
+   * Extracts key, values from input string using ',' as delimiter1 and '=' as delimiter2
+   * and creates a new map column.
+   * @group normal_funcs
+   * @since 2.1.0
+   */
+  def str_to_map(e: Column): Column = withExpr { new StringToMap(e.expr) }
```

Comment:
For now let's not register the hive fallback functions here, we will think about it later.